### PR TITLE
fix(api.js) Replace `number[]`with `Uint8Array`. fixes #3306

### LINKED ISSUE #3306: Replace `number[]` with `Uint8Array`
Our api functions that deal with binary data currently have the type `number[]` which serializes to the correct JSON, but has sub-par DX when used with other JavaScript tools and libraries. 
The attached PR replaces all occurrences of `number[]` with `Uint8Array` the correct `Vec<u8>` equivalent.

This is a breaking change though, as functions no longer accept plain number arrays. Typescript users will get correct errors, but non-ts users will receive `invalid type: map, expected a sequence`

This change will only affect the user facing interface, as Uint8Arrays will be converted into `number[]` under the hood to get  preserve the correct serialization

### PR DESCRIPTION
**Pull request by JonasKruckenberg**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

attacked issue #3306

- [ ] Yes, and the changes were approved in issue #___
- [ ] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

**Comment by FabianLars** [comment]
if we merge this we need a changefile too :P
ngl, i'm glad i'm not the only one who always forgets them 😂

**Comment by JonasKruckenberg** [comment]
> ngl, i'm glad i'm not the only one who always forgets them 😂

Throwback to the time I merged a PR **and** deleted the branch without a changefile and had to open a super embarrassing PR with only a changefile 😂 

Edit:
> if we merge this we need a changefile too :P

After your comment on the issue, I feel like making this a patch thing is good enough 🤔 

Edit2: Not that it matters much, this will result in a major release anyway lol

**Comment by FabianLars** [comment]
yeah we did `patch` all the time, for pretty much every change. Really doesn't seem to matter for pre-releases, or at least from 0.x to 1.0-pre, whatever.

File: tooling/api/src/http.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@ enum ResponseType {
   Binary = 3
 }
 
-type Part = string | number[]
+type Part = string | Uint8Array
```
Comment (FabianLars) [inline tooling/api/src/http.ts:40]: I think this causes issues at the form() function on line 60 (can't add the comment there)

**Review by FabianLars — CHANGES_REQUESTED** [review]

File: tooling/api/src/http.ts (hunk 1 above)
Comment (JonasKruckenberg) [reply tooling/api/src/http.ts:40]: Yikes, yeah good catch. I think I'll add an internal `serializePayload` function or something.

File: tooling/api/src/fs.ts [hunk 2]
```diff
@@ -119,15 +119,17 @@ async function readTextFile(
 async function readBinaryFile(
   filePath: string,
   options: FsOptions = {}
-): Promise<number[]> {
-  return invokeTauriCommand<number[]>({
+): Promise<Uint8Array> {
+  const arr = invokeTauriCommand<number[]>({
```
Comment (amrbashir) [inline tooling/api/src/fs.ts]:
```suggestion
  const arr = await invokeTauriCommand<number[]>({
```

**Pull request merged**